[tflite] fix missing dep for preprocess_coco_minival

freedomtan opened the pull request:
add missing dep for
```
//tensorflow/lite/tools/evaluation/tasks/coco_object_detection:preprocess_coco_minival
```

`preprocess_coco_minival` uses `evaluation_stages`. `evaluation_stages` uses
`preprocessing_steps` which is not generated.

Add the build rule and corresponding dependency so that
```
bazel run //tensorflow/lite/tools/evaluation/tasks/coco_object_detection:preprocess_coco_minival -- \
  --images_folder=/path/to/val2014 \
  --instances_file=/path/to/instances_val2014.json \
  --whitelist_file=/path/to/minival_whitelist.txt \
  --output_folder=/path/to/output/folder
```
will work again.

freedomtan committed:
[tflite] fix missing dep for preprocess_coco_minival

add missing dep for
```
//tensorflow/lite/tools/evaluation/tasks/coco_object_detection:preprocess_coco_minival
```

`preprocess_coco_minival` uses `evaluation_stages`. `evaluation_stages` uses
`preprocessing_steps` which is not generated.

Add the build rule and corresponding dependency so that
```
bazel run //tensorflow/lite/tools/evaluation/tasks/coco_object_detection:preprocess_coco_minival -- \
  --images_folder=/path/to/val2014 \
  --instances_file=/path/to/instances_val2014.json \
  --whitelist_file=/path/to/minival_whitelist.txt \
  --output_folder=/path/to/output/folder
```
will work again.

renjie-liu commented:
Hi can you sync to head and try again? I cannot reproduce the error at head

freedomtan commented:
I can reproduce the error with cf50b1f on Ubuntu 18.04 and macOS 10.15.x
Note that this is a run time dependency problem.
```
> git checkout master
> git pull
> bazel clean --expunge
> bazel run --config opt \
  //tensorflow/lite/tools/evaluation/tasks/coco_object_detection:preprocess_coco_minival -- \
  --images_folder=/hack/freedom/mscoco/val2014 \
  --instances_file=/tmp/instances_val2014.json \
  --whitelist_file=/hack/freedom/mscoco/mscoco_minival_ids.txt \
  --output_folder=/hack/freedom/mscoco/processed

...
INFO: Found 1 target...
Target //tensorflow/lite/tools/evaluation/tasks/coco_object_detection:preprocess_coco_minival up-to-date:
  bazel-bin/tensorflow/lite/tools/evaluation/tasks/coco_object_detection/preprocess_coco_minival
INFO: Elapsed time: 91.072s, Critical Path: 22.86s
INFO: 304 processes: 304 local.
INFO: Build completed successfully, 313 total actions
INFO: Running command line: bazel-bin/tensorflow/lite/tools/evaluation/tasks/coco_object_detection/preprocess_coco_minival '--images_folder=/hack/freedom/mscoco/val2014' '--instances_file=/tmp/instances_val2014.json' '--whitelist_file=/hack/freedom/mscoco/mscoco_minival_ids.txt' '--output_folder=/hack/freedom/mscoco/prINFO: Build completed successfully, 313 total actions
Traceback (most recent call last):
  File "/home/freedom/.cache/bazel/_bazel_freedom/6c0437fbbcac51299f8083154d82c985/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/lite/tools/evaluation/tasks/coco_object_detection/preprocess_coco_minival.runfiles/org_tensorflow/tensorflow/lite/tools/evaluation/tasks/coco_object_detection/preprocess_coco_minival.py", line 39, in <module>
    from tensorflow.lite.tools.evaluation.proto import evaluation_stages_pb2
  File "/home/freedom/.cache/bazel/_bazel_freedom/6c0437fbbcac51299f8083154d82c985/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/lite/tools/evaluation/tasks/coco_object_detection/preprocess_coco_minival.runfiles/org_tensorflow/tensorflow/lite/tools/evaluation/proto/evaluation_stages_pb2.py", line 16, in <module>
    from tensorflow.lite.tools.evaluation.proto import preprocessing_steps_pb2 as tensorflow_dot_lite_dot_tools_dot_evaluation_dot_proto_dot_preprocessing__steps__pb2
ImportError: cannot import name 'preprocessing_steps_pb2'
```

rthadur commented on tensorflow/lite/tools/evaluation/tasks/coco_object_detection/BUILD:
```diff
@@ -13,7 +13,10 @@ py_binary(
     python_version = "PY3",
```
@freedomtan here is the internal error , can you please check 

ERROR: lite/tools/evaluation/tasks/coco_object_detection/BUILD:11:10: no such target '/lite/tools/evaluation/proto:preprocessing_steps_py': target 'preprocessing_steps_py' not declared in package 'lite/tools/evaluation/proto' defined by lite/tools/evaluation/proto/BUILD and referenced by 'lite/tools/evaluation/tasks/coco_object_detection:preprocess_coco_minival'

freedomtan replied:
@rthadur That is a known issue. I sent a PR (#38833) for that. Since @multiverse-tf has a better fix, I closed my PR. Hopefully, that will be fixed soon.

freedomtan replied:
rebased to master and removed the `proto:preprocessing_steps` I added because it was added by other commit already

rthadur replied:
@freedomtan still failing same error , can you please check

freedomtan replied:
close this, d76bbee1341 already fixed the problem :-)

freedomtan committed:
rebase and remove proto:preprocessing_steps

rebase to master and remove my
`lite/tools/evaluation/proto:preprocessing_steps` because it was
added in master branch.
